Services/WatchlistService.swift

```diff
@@ -6,76 +6,129 @@
     }
     
-    func toggleStatus(for item: WatchlistItem)-> WatchlistStatus {
-        var watchlist = getWatchlist()
-        
-        let currentStatus: WatchlistStatus = watchlist[item.id] != nil ? .added : .notAdded
-        switch currentStatus {
-        case .added:
-            watchlist.removeValue(forKey: item.id)
-            return saveWatchlist(watchlist) ? .notAdded : .added
-        case .notAdded:
-            watchlist[item.id] = item
-            return saveWatchlist(watchlist) ? .added : .notAdded
+    var errorImage: UIImage? {
+        switch self {
+        case .failedToFetchFromPersistentStore:
+            return UIImage(named: "dizzy")
         }
     }
+}
+
+class Container {
+    static let shared: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: "WatchlistMovies")
+        container.viewContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
+        container.loadPersistentStores {storeDescription, error in
+            if let error = error {
+                print("Unresolved error \(error)")
+            }
+        }
+        return container
+    }()
     
-    func getWatchlistItems() -> [WatchlistItem] {
-        let watchlist = getWatchlist()
+    private init() {
+    }
+}
+
+class WatchlistService: WatchlistServiceProtocol {
+    let container = Container.shared
+    var predicate: NSComparisonPredicate?
+
+    func getStatus(for id: Int) throws -> WatchlistStatus {
+        let request = WatchlistMovie.createFetchRequest()
+        predicate = NSComparisonPredicate(format: "id == %@", id as NSNumber)
+        request.predicate = predicate
         
-        var items = [WatchlistItem]()
-        for value in watchlist.values {
-            items.append(value)
+        do {
+            let movies = try container.viewContext.fetch(request)
+            
+            if !movies.isEmpty {
+                return .added
+            } else {
+                return .notAdded
+            }
+        } catch {
+            print("Fetch failed")
+            throw WatchlistServiceError.failedToFetchFromPersistentStore
         }
-
-        return items
     }
     
-    private func getWatchlist() -> Watchlist {
-        let defaults = UserDefaults.standard
+    func toggleStatus(for item: WatchlistMovieConfiguration) -> WatchlistStatus {
+        let request = WatchlistMovie.createFetchRequest()
+        predicate = NSComparisonPredicate(format: "id == %@", item.id as NSNumber)
+        request.predicate = predicate
         
-        if let watchlistData = defaults.object(forKey: WatchlistService.watchlistKey) as? Data {
-            let decoder = JSONDecoder()
-            if let watchlist = try? decoder.decode(Watchlist.self, from: watchlistData) {
-                return watchlist
+        do {
+            let movies = try container.viewContext.fetch(request)
+            
+            if !movies.isEmpty {
+                container.viewContext.delete(movies[0])
```

Comment:
Just another way to write the same code without accessing the array directly (nothing wrong with your code!)
```suggestion
            if let movie = movies.first {
                container.viewContext.delete(movie)
```